src/GSoft.Extensions.MediatR.Analyzers/NamingConventionAnalyzer.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System.Collections.Immutable;
+using GSoft.Extensions.MediatR.Analyzers.Internals;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace GSoft.Extensions.MediatR.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class NamingConventionAnalyzer : DiagnosticAnalyzer
+{
+    internal static readonly DiagnosticDescriptor UseCommandOrQuerySuffixRule = new DiagnosticDescriptor(
+        id: RuleIdentifiers.UseCommandOrQuerySuffix,
+        title: "Name should end with Command or Query",
+        messageFormat: "Name should end with Command or Query",
```

Comment:
For consistency sake, for both `title` and `messageFormat` parameters, be sure to use single quotes around verified "tokens" such as `Command` and `Query` like such:
```suggestion
        title: "Name should end with 'Command' or 'Query'",
        messageFormat: "Name should end with 'Command' or 'Query'",
```

Reply:
🕵️ Nice catch, I might have loose some changes in the process.